cpp/benchmarks/hashing/hash.cpp

```diff
@@ -54,11 +54,11 @@ static void BM_hash(benchmark::State& state, cudf::hash_id hid, contains_nulls h
 #define HASH_BENCHMARK_DEFINE(hid, n) H_BENCHMARK_DEFINE(concat(hid, _, n), hid, n)
 
 HASH_BENCHMARK_DEFINE(HASH_MURMUR3, nulls)
-HASH_BENCHMARK_DEFINE(HASH_MD5, nulls)
 HASH_BENCHMARK_DEFINE(HASH_SERIAL_MURMUR3, nulls)
 HASH_BENCHMARK_DEFINE(HASH_SPARK_MURMUR3, nulls)
+HASH_BENCHMARK_DEFINE(HASH_MD5, nulls)
```

Comment:
A good chunk of the diffs in this PR are just from moving code around. I put MD5 at the bottom of the list of hashing functions defined in the `hash_id` enum, and reorganized everything else to have the same order. That means that the MD5 and SHA features can be listed next to each other in a subsequent PR #9215. MD5 and SHA are in the same family of cryptographic hash functions, so it's a logical grouping.